src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -394,10 +349,27 @@ void FileIOMicroReadBenchmarkFixture::pread_atomic_random_multi_threaded(benchma
   close(fd);
 }
 
-void FileIOMicroReadBenchmarkFixture::aio_single_threaded(benchmark::State& state) {
+void create_aio_request(struct aiocb &request, int const fd, off_t const offset, volatile void* buffer, size_t const bytes, int const aio_lio_opcode) {
+  memset(&request, 0, sizeof(request));
+
+  request.aio_fildes = fd;
+  request.aio_offset = offset;
+  request.aio_buf = buffer;
+  request.aio_nbytes = bytes;
+  request.aio_lio_opcode = aio_lio_opcode;
+}
+
+void FileIOMicroReadBenchmarkFixture::aio_sequential_read_single_threaded(benchmark::State& state) {
```

Comment:
This is single-threaded, but also mimics a blocking behavior as no further calls are issues. So you issue one call and then suspend until it's done. This is not asynchronous I/O.

Reply:
Since I don't see a value in having a pseudo-blocking async implementation, I'd consider your multi-threaded implementation to be the actual single-threaded implementation.